library/train_util.py

```diff
@@ -1071,14 +1075,18 @@ def __getitem__(self, index):
                 crop_ltrb = image_info.latents_crop_ltrb  # calc values later if flipped
                 if not flipped:
                     latents = image_info.latents
+                    mask = image_info.mask
                 else:
                     latents = image_info.latents_flipped
+                    mask = image_info.mask_flipped
 
                 image = None
             elif image_info.latents_npz is not None:  # FineTuningDatasetまたはcache_latents_to_disk=Trueの場合
                 latents, original_size, crop_ltrb, flipped_latents = load_latents_from_disk(image_info.latents_npz)
+                mask = load_mask(image_info.absolute_path, image_info.resized_size) / 225
```

Comment:
This should probably be 255?

Of note: loading masks breaks when loading a latent that was cached to disk if the original image was resized and cropped, because the loaded mask doesn't undergo the same load/crop routine, and ends up mismatched to the latent. When loading latents from disk, the original image is never considered. I think that the current implementation of `trim_and_resize_if_required` prevents replication of the original crop if random_crop is specified, since the coordinates of the crop are not passed back; crop_ltrb is something other than the coords actually used for cropping the image.

I've solved this in my local copy by just persisting the mask in the .npz, but that obviously blows up the cache size.

Reply:
Whoops! nice catch.

I guess right now it is a limitation - I don't see a reliable way of preserving the crop parameters other than serializing the mask like you did (or a latent sized crop), or at least embed the crop parameters into the npz.

Reply:
Keeping, returning, persisting, and retrieving a second set of crop parameters from `trim_and_resize_if_required` would be the much lighter option. Either way, a change to the npz format is likely necessary, or latent disk caching should fail loudly when using masked loss and images which need cropping.